bin/src/data/transform/data_transformation_generators.py

```diff
@@ -0,0 +1,139 @@
+"""
+This file contains noise generators classes for generating various types of noise.
+"""
+
+from abc import ABC, abstractmethod
+from typing import Any
+import numpy as np
+import multiprocessing as mp
+
+
+class AbstractDataTransformer(ABC):
+    """
+    Abstract class for data transformers. 
+    All data transformers should have the seed in it. Because the multiprocessing of them could unset the seed in short.
+    """
+
+    def __init__(self):
+        self.modify_column = None
+        self.add_row = None
+        self.training_data_only = False
```

Comment:
Add a comment with TODO find better way to specify set

Reply:
also if not too much hustle merge the modify_column and add_row into one flag.

Reply:
done